masonry/src/render_root.rs

```diff
@@ -570,8 +570,9 @@ impl RenderRoot {
             self.state.focused_widget = new;
             self.root_lifecycle(event);
 
-            // TODO - Handle IME
-            // Send TextFieldFocused(focused_widget) signal
+            self.state
+                .signal_queue
+                .push_back(RenderRootSignal::TextFieldFocused);
```

Comment:
The code shouldn't unconditionally send a TextFieldFocused event. There are cases where `new` is `None`, or refers to a non-textbox widget.

Reply:
I handle this now, thank you.